### ISSUE TITLE
Can't increase ASIO buffer beyond 256

### ISSUE THREAD

**Issue by paulheu** [eiz/SynchronousAudioRouter#40]
On my  RME HDSP 3296 when running SAR 0.13 in Reaper I am unable to use ASIO buffer sizes over 256. When I try to set 512 I get an error on the ASIO device in Reaper. If I use the RME drivers on their own this works just fine.

Tried un- and reinstall of 0.13, went back to 0.12 all to no avail
Running fully updated WIN10 pro

**Comment by eiz** [comment]
What sample rate? At 48khz 512 samples would be greater than the audio engine periodicity and result in glitches.

**Comment by paulheu** [comment]
I am running at 48K, the RME drivers will allow me to set and use 512 samples and I am not encountering any issues. Problem I am facing is that audio is giving me crackles at 256 samples, specifically when running certain games on the 4K monitor and then opening for instance a browser or something else on one of my two 1080p monitors. When I do not run SAR I'm not having this issue. I want to try and run at 512 samples to see if that makes a difference as the latency is not an issue in my situation anyway. I guess I will try at 44K when I get back home this week..

**Comment by paulheu** [comment]
So, tried this with 44.1K samplerate and get same result. anything over 256 buffer will fail when using SAR, when using the RME drivers on their own I have no issues with this.

**Comment by eiz** [comment]
Yeah, unfortunately the tradeoff SAR makes to get minimum latency does mandate that the buffer size be less than 10ms worth of samples. If you're experiencing crackling, make sure your audio software is running at real time priority and consider checking your system with LatencyMon or another DPC latency tester. If that doesn't help, I'm sorry to say you may need to find some other solution.

To give way too much detail, there are 2 sources of periodic events happening, the ASIO tick (driven by the audio hardware's sample clock normally) and the Windows audio engine's timer (every 10ms driven by a system timer interrupt). Increasing the ASIO buffer size slows the frequency of ASIO tick events and if it becomes slower than the Windows audio engine's frequency, because of the way SAR integrates with WaveRT this will cause bad buffer underruns. To prevent this scenario, SAR currently does not allow buffer sizes greater than the sample rate divided by 100. I don't have an immediate fix for it; though I'm sure it's possible to resolve, it just hasn't been a high priority as low latency was the whole point of writing it for me.
